BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -137,7 +141,11 @@ uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
 void CAN1_RX0_IRQHandler(void)
 {
     CAN_Receive(CAN1, CAN_FIFO0, &RxMessage);
-
+    for(int i = 0; i < 8; i++){
+        CANFIFO[CANFIFOHEAD][i] = RxMessage.Data[i]; //store data in FIFO
+        (++CANFIFOHEAD)%CANFIFOMAX; //increment FIFO Head
```

Comment:
This does not do what you think.

Reply:
I think this increments the row of the matrix and mods it by the max so it wraps around when done. I moved the incrementing of the head outside the for loop. Would that work or is there something else I am missing.

Reply:
Specifically, the two lines shown below are not equivalent.
```c
(++CANFIFOHEAD)%CANFIFOMAX;

CANFIFOHEAD = (CANFIFOHEAD + 1) % CANFIFOMAX;
```

Reply:
Changed this everywhere I did it.